doc/pick_place/src/pick_place_tutorial.cpp

```diff
@@ -1,244 +1,215 @@
   std::vector<moveit_msgs::Grasp> grasps;
-
-  geometry_msgs::PoseStamped p;
-  p.header.frame_id = "base_footprint";
-  p.pose.position.x = 0.34;
-  p.pose.position.y = -0.7;
-  p.pose.position.z = 0.5;
-  p.pose.orientation.x = 0;
-  p.pose.orientation.y = 0;
-  p.pose.orientation.z = 0;
-  p.pose.orientation.w = 1;
-  moveit_msgs::Grasp g;
-  g.grasp_pose = p;
-
-  g.pre_grasp_approach.direction.vector.x = 1.0;
-  g.pre_grasp_approach.direction.header.frame_id = "r_wrist_roll_link";
-  g.pre_grasp_approach.min_distance = 0.2;
-  g.pre_grasp_approach.desired_distance = 0.4;
-
-  g.post_grasp_retreat.direction.header.frame_id = "base_footprint";
-  g.post_grasp_retreat.direction.vector.z = 1.0;
-  g.post_grasp_retreat.min_distance = 0.1;
-  g.post_grasp_retreat.desired_distance = 0.25;
-
-  openGripper(g.pre_grasp_posture);
-
-  closedGripper(g.grasp_posture);
-
-  grasps.push_back(g);
-  group.setSupportSurfaceName("table");
-  group.pick("part", grasps);
+  grasps.resize(1);
+
+  // Setting the grasp pose
+  grasps[0].grasp_pose.header.frame_id = "panda_link0";
+  grasps[0].grasp_pose.pose.orientation =
+      tf::createQuaternionMsgFromRollPitchYaw(-1.5707963267948966, -0.7853981633974483, -1.5707963267948966);
+  // This is the pose of panda_link8.
+  // From panda_link8 to the palm of the eef the distance is 0.058, the cube starts 0.01 before 5.0 (half of the length
+  // of the cube).
+  // Therfore, the posotion for panda_link8 = 5 - (length of cube/2 - distance b/w panda_link8 and palm of eef - some
+  // extra padding
+  grasps[0].grasp_pose.pose.position.x = 0.415;
+  grasps[0].grasp_pose.pose.position.y = 0;
+  grasps[0].grasp_pose.pose.position.z = 0.5;
+
+  // Setting the pre_grasp_approach with direction as positive x axis.
+  grasps[0].pre_grasp_approach.direction.vector.x = 1.0;
+  grasps[0].pre_grasp_approach.direction.header.frame_id = "panda_link0";
+  grasps[0].pre_grasp_approach.min_distance = 0.095;
+  grasps[0].pre_grasp_approach.desired_distance = 0.115;
+
+  // Setting the post_grasp_retreat with direction as positive z axis.
+  grasps[0].post_grasp_retreat.direction.header.frame_id = "panda_link0";
+  grasps[0].post_grasp_retreat.direction.vector.z = 1.0;
+  grasps[0].post_grasp_retreat.min_distance = 0.1;
+  grasps[0].post_grasp_retreat.desired_distance = 0.25;
+
+  // Set the posture of the eef before we start the grasp.
+  openGripper(grasps[0].pre_grasp_posture);
+  // END_SUB_TUTORIAL
+
+  // BEGIN_SUB_TUTORIAL pick2
+  // Set the posture of the eef for the grasp
+  closedGripper(grasps[0].grasp_posture);
+  // END_SUB_TUTORIAL
+
+  // BEGIN_SUB_TUTORIAL pick3
+  // Set support surface as table1.
+  move_group.setSupportSurfaceName("table1");
+  // Call pick to pick up the object using the grasps given
+  move_group.pick("object", grasps);
+  // END_SUB_TUTORIAL
 }
 
 void place(moveit::planning_interface::MoveGroupInterface& group)
```

Comment:
When I run locally place fails with "All supplied place locations failed. Retrying last location in verbose mode."

Reply:
It happens here also rarely. I am guessing its because there is only one grasp given and it only tries to plan once.

Reply:
Add a comment that this is a known issue with a TODO to fix.

Reply:
Would generating multiple grasps fix this?